Sources/Utility/ProcessGroup.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Dispatch
+import Foundation
+
+/// A process group is a small warpper for collection of processes.
+/// 
+/// This class is thread safe.
+public final class ProcessGroup {
+
+    /// Array to hold the processes.
+    private var processes: [Process] = []
+
+    /// Queue to mutate internal states of the process group.
+    private let serialQueue = DispatchQueue(label: "org.swift.swiftpm.processgroup")
+
+    public init() {}
+
+    /// Add a process to the process group.
+    public func add(_ process: Process) {
```

Comment:
I figured we can expand this to work with our system and popen calls if needed in future (create another add method for them).
I used Process here because it felt like duplicating efforts (atleast for now)